### PR TITLE
Drop the string initialization of Tree

### PR DESCRIPTION
**Pull request by SocialfiPanda**
This is the pull request for fixing issue #675.

### CODE REVIEW

File: nltk/test/tree.doctest [hunk 1]
```diff
@@ -39,39 +39,6 @@ tree object to one of several standard tree encodings:
             [.dp [.d the ] [.np dog ] ]
             [.vp [.v chased ] [.dp [.d the ] [.np cat ] ] ] ]
 
-Trees can be initialized from treebank strings:
-
-    >>> tree2 = Tree('(S (NP I) (VP (V enjoyed) (NP my cookie)))')
-    >>> print(tree2)
-    (S (NP I) (VP (V enjoyed) (NP my cookie)))
-
-Trees can be compared for equality:
```
Comment (kmike) [inline nltk/test/tree.doctest:48]: Why drop comparison tests?

File: nltk/tree.py [hunk 2]
```diff
@@ -51,21 +51,10 @@ class Tree(list):
         (VP (V saw) (NP him))
         >>> print(s[1,1])
         (NP him)
-        >>> t = Tree("(S (NP I) (VP (V saw) (NP him)))")
-        >>> s == t
-        True
-        >>> t[1][1].set_label('X')
```
Comment (kmike) [inline nltk/tree.py:57]: This test showed how to work with Tree labels and how to move subtrees, why remove it?

File: nltk/tree.py [hunk 3]
```diff
@@ -85,28 +74,21 @@ class Tree(list):
     specifying ``tree[i]``; or a sequence *i1, i2, ..., iN*,
     specifying ``tree[i1][i2]...[iN]``.
 
-    Construct a new tree.  This constructor can be called in one
-    of two ways:
+    Construct a new tree.  This constructor can be called in the following way:
```
Comment (kmike) [inline nltk/tree.py]: What do you think about referring to `fromstring` method here? Something like "You can also construct a tree from a string; use :meth:fromstring for that".

File: nltk/test/tree.doctest (hunk 1 above)
Comment (SocialfiPanda) [inline nltk/test/tree.doctest:48]: The issue #675 asks for dropping the string initialization of the Tree. So, these two still work:

Tree('dp', [Tree('d', ['the']), Tree('np', ['dog'])])
Tree.fromstring('(dp (d the) (np dog))')

But this not:
Tree('(dp (d the) (np dog))')

If I understand the issue #675 correctly, after the pull request, we aren't supposed to be able to initialize a Tree by supplying it a string (line 44 should break).

Comment (kmike) [inline nltk/test/tree.doctest:48]: Right; can we replace `Tree('(dp (d the) (np dog))')` with `Tree.fromstring('(dp (d the) (np dog))')` here and keep the tests?

Comment (kmike) [inline nltk/test/tree.doctest:48]: I'm not asking to keep tests in lines 44-46, the question was about lines 50-...

Comment (SocialfiPanda) [inline nltk/test/tree.doctest:48]: can we replace Tree('(dp (d the) (np dog))') with Tree.fromstring('(dp (d the) (np dog))') here and keep the tests?
- Sure. I will do that.

File: nltk/tree.py (hunk 2 above)
Comment (SocialfiPanda) [inline nltk/tree.py:57]: Again, now we can't initialize a tree by a string. I can change replace 
replace Tree("(S (NP I) (VP (V saw) (NP him)))") with Tree.fromstring("(S (NP I) (VP (V saw) (NP him)))") and keep line 57 for setting the label. You think it's good?

File: nltk/tree.py (hunk 3 above)
Comment (SocialfiPanda) [inline nltk/tree.py]: I think it sounds good. I will add that to the docstring.

Also, I just found out there were many places in nltk/tree.py still using old string initilization syntax. e.g. line 212:

">>> t = Tree('(S (NP (D the) (N dog)) (VP (V chased) (NP (D the) (N cat))))')"

Sorry I forget to change these. I will replace them all with Tree.fromstring instead, and make a new commit addressing all your concerns above. Really appreciate your help!

**Comment by SocialfiPanda** [comment]
These two last commits add back the test cases I deleted in my previous commits. Also, it changes:

Tree(str(tree)) to Tree.fromstring(str(tree)) in the docstring of the tree.py and in the test cases of test/tree.doctest.

**Pull request merged**